### DISCUSSION TITLE
CORS Origins Regex Parsing?

### DISCUSSION THREAD

**Discussion by Priyansh4444** [honojs/hono#3709] [Ideas]
When CORS origins takes a string[], does it do regex parsing to found the allowed origins? and if it doesn't should it?

I was mentoring for a hackathon, and he was getting CORS errors, since the list had [ '*' ] as one of them, so would it be possible to make that happen?
I'd love to contribute and spin this up if this was an intended feature, but i am unsure if allowing regex in the list of strings is supposed to be allowed!

**Comment by Mudbill** [comment]
I don't think strings should implicitly be parsed as regexes. I think a better approach would be to allow RegExp objects alongside strings as input. E.G: `["https://public.website", /localhost:\d*/]`

Though you could do an equivalent with the callback:

```ts
cors({
    origin: (origin) => {
        if (/localhost:\d*/.test(origin)) {
            return origin
        }
        return "https://public.website"
    }
})
```
